osquery/tables/system/linux/portage.cpp

```diff
@@ -0,0 +1,416 @@
+        value = i.substr(eqp + 1);
+        boost::algorithm::trim_all(value);
+      } else {
+        value += " " + i;
+      }
+
+      if (boost::starts_with(value, "\"") || boost::starts_with(value, "'")) {
+        value = value.erase(0, 1);
+        boost::algorithm::trim_all(value);
+      }
+
+      if (boost::ends_with(value, "\\")) {
+        keep_building = true;
+        value = value.erase(value.length() - 1, 1);
+        boost::algorithm::trim_all(value);
+      } else {
+        keep_building = false;
+      }
+
+      if (boost::ends_with(value, "\"") || boost::ends_with(value, "'")) {
+        value = value.erase(value.length() - 1, 1);
+        boost::algorithm::trim_all(value);
+      }
+
+      if (keep_building) {
+        continue;
+      }
+
+      Row r;
+      r["key"] = key;
+      r["value"] = value;
+      results.push_back(r);
+    }
+  }
+
+  return results;
+}
+
+QueryData parsePortageUseContent(const std::string& content) {
+  QueryData results;
+
+  for (const auto& i : osquery::split(content, "\n")) {
+    auto line = split(i);
+    if (line.size() == 0 || boost::starts_with(line[0], "#")) {
+      continue;
+    }
+    Row r;
+    std::pair<std::string, std::string> package =
+        portageSplitPackageVersion(line[0]);
+    r["package"] = package.first;
+    r["version"] = package.second;
+    if (line.size() > 1) {
+      std::vector<std::string> use_on;
+      std::vector<std::string> use_off;
+      for (size_t i = 1; i < line.size(); ++i) {
+        if (boost::starts_with(line[i], "#")) {
+          break;
+        }
+        if (!boost::starts_with(line[i], "-")) {
+          use_on.push_back(line[i]);
+        } else {
+          use_off.push_back(line[i].erase(0, 1));
+        }
+      }
+      r["use_on"] = boost::algorithm::join(use_on, " ");
+      r["use_off"] = boost::algorithm::join(use_off, " ");
+    }
+    results.push_back(r);
+  }
+
+  return results;
+}
+
+/* Functions refered from tables */
+QueryData portagePackages(QueryContext& context) {
+  if (isDirectory(portagePackageDir)) {
+    std::string conetntUse;
+    readFile(portageWorld, conetntUse).ok();
+    return parsePortagePackages(portagePackageDir, conetntUse);
+  } else {
+    return {};
+  }
+}
+
+QueryData genPortageMake(QueryContext& context) {
+  std::string content;
+  if (readFile(portageEtcMake, content).ok()) {
+    return parsePortageMakeContent(content);
+  } else if (readFile(portageEtcPortageMake, content).ok()) {
+    return parsePortageMakeContent(content);
+  } else {
+    return {};
+  }
+}
+
+QueryData genPortageUse(QueryContext& context) {
+  std::string content;
+  if (readFile(portageUse, content).ok()) {
+    return parsePortageUseContent(content);
+  } else {
+    return {};
+  }
+}
+
+QueryData genPortageKeywordSummary(QueryContext& context) {
+  std::string keywords;
+  std::string masked;
+  std::string unmasked;
+  bool keywordsResult = readFile(portageKeywords, keywords).ok();
```

Comment:
Instead of creating intermediate booleans, you could have the conditional check `!.empty()` for each.